pkg/controllers/nodeclaim/inplaceupdate/suite_test.go

```diff
@@ -190,6 +192,192 @@ var _ = Describe("Unit tests", func() {
 			// Should be different because NodeClass overrides AdditionalTags
 			Expect(hash1).ToNot(Equal(hash2))
 		})
+
+		It("should produce different hashes for AKS machines vs VMs", func() {
+			options := test.Options()
+			options.AdditionalTags = map[string]string{
+				"test-tag": "test-value",
+			}
+			options.NodeIdentities = []string{
+				"/subscriptions/1234/resourceGroups/mcrg/providers/Microsoft.ManagedIdentity/userAssignedIdentities/myid1",
+			}
+
+			// Regular VM NodeClaim
+			vmNodeClaim := coretest.NodeClaim()
+			hash1, err := inplaceupdate.HashFromNodeClaim(options, vmNodeClaim, nodeClass)
+			Expect(err).ToNot(HaveOccurred())
+
+			// AKS Machine NodeClaim
+			aksMachineNodeClaim := coretest.NodeClaim()
+			aksMachineNodeClaim.Annotations = map[string]string{
+				v1beta1.AnnotationAKSMachineResourceID: "/subscriptions/123/resourceGroups/rg/providers/Microsoft.ContainerService/managedClusters/cluster/agentPools/pool/machines/test-machine",
+			}
+			hash2, err := inplaceupdate.HashFromNodeClaim(options, aksMachineNodeClaim, nodeClass)
+			Expect(err).ToNot(HaveOccurred())
+
+			// Should be different because AKS machines don't include identities in hash
+			Expect(hash1).ToNot(Equal(hash2))
+		})
+	})
+
+	Context("calculateAKSMachinePatch", func() {
```

Comment:
minor: Curious why this machine stuff is here, but other machine stuff is in new test file? Could put this in that file too?